ivy.el

```diff
@@ -3227,12 +3227,29 @@ Otherwise, ~/ will move home."
 
 (defvar ivy--exhibit-timer nil)
 
+(defalias 'ivy-input-changed-p
+  (let (last-len)
+    (lambda ()
+      (let ((len (length ivy-text))
+            (more-chars-len (ivy-alist-setting ivy-more-chars-alist)))
+        (prog1 (cond ((< len more-chars-len)
+                      ;; force ui update, changed
+                      nil)
+                     (t ;; changed if len is different
+                      (not (eql len last-len))))
```

Comment:
Can this `cond` be simplified as follows?
```el
(and (>= len more-chars-len)
     (not (eql len last-len)))
```

Is `more-chars-len` guaranteed to be non-`nil` here?

Reply:
If the user has set `ivy-more-chars-alist` to not have a t pair, then no.

Reply:
But perhaps a safe guard of value 0 can be added 
```
(let (... 
       (more-chars-len (or (ivy-alist-setting ivy-more-chars-alist)
                                     0))
    ...)
```

Reply:
Having a safe guard SGTM.